### ISSUE TITLE
UIView saved image can't stitch together but after Photoshop resave they can?

### ISSUE THREAD

**Issue by shawhu** [foundry/OpenCVSwiftStitch#12]
I'm working on a project that uses your library.

I tried to save the captured photos to the filesystem and later reload them when stitching. I tried several times, sometimes it worked, but most of the time it doesn't, The stitching just asking for 500MB extra memory. And then I did a test, I exported the images saved thru itunes file sharing and I resaved them AS IS using photoshop, and then I drag them back to the Images.xcassets. And then I realized, photoshop saved files work! 100% of time.

And then I tried to drag the exported photos originals back to the Images.xcassets and see if they work as well, they don't, just like the one saved in the mydocument.

I've uploaded the original photos as well as photoshop saved copies here
https://www.dropbox.com/s/2srfpi57j99e0xr/Archive.zip?dl=0

When you have time, I hope you can take a look and just drag those files to the images asset and try to stitch them. I think it must be something that I can somehow save those photos with different or some extra steps so they can be used by the stitching process

**Comment by foundry** [comment]
Hello shawhu. Thanks for taking the time to raise this issue, which I was already aware of but had not correctly diagnosed. 

The cause of the problem is Apple's use of the `imageOrientation` flag in `UIImage` to display images at the correct orientation. All images taken by iOS cameras are saved natively in landscapeLeft orientation (as if you had taken the photo in landscape with the home button on the right). They are transformed to the correct orientation for display only. openCV on the other hand expects you to feed it images in the correct pixel orientation. 

When you re-save an image using Photoshop, it performs the pixel rotation for you, so openCV will work fine with those re-saved images.

Now I have added a category on UIImage to check the orientation and rotate the actual pixels if necessary before sending to openCV. I am using the category in the wrapper, not in the swift viewController, so I have implemented in objective-C. I have tested this with your samples and it works fine. 

Regarding [your other issue](https://github.com/foundry/OpenCVSwiftStitch/issues/11), I will reply later when I have more time.
